### PR TITLE
PERF: Do not import torch to reduce itk import time

### PR DESCRIPTION
**Pull request by tbaudier**
https://discourse.itk.org/t/torch-import-time/6354

Torch takes time to import. In types.py torch is imported to check if it's present or not in the environment, but it's never used.

Now, torch is detected with importlib.metadata.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Wrapping/Generators/Python/itk/support/types.py [hunk 1]
```diff
@@ -33,10 +34,10 @@
     pass
 _HAVE_TORCH = False
 try:
-    import torch
+    metadata('torch')
 
     _HAVE_TORCH = True
-except ImportError:
+except importlib.metadata.PackageNotFoundError:
```
Comment (thewtex) [inline Wrapping/Generators/Python/itk/support/types.py:40]: Following the CI errors, we may also need `import importlib` at the top.

**Review by thewtex — CHANGES_REQUESTED** [review]
@tbaudier thank you for the contribution! :pray: 

Please see a comment inline that could help with the CI failures.

Could also please apply the same method to the other packages in the same file?

**Comment by tbaudier** [comment]
Sorry, I committed the src files but my modified files were in the built folder. Now it's ok.

I tested this code and it worked well:
```py
import torch
import numpy as np
a = np.random.rand(10,10)
t = torch.from_numpy(a)
smoothed = itk.median_image_filter(a, radius=2)
```

According to your advise, I did the modification for xarray too but I did not test it.

For numpy ArrayType, I changed anything because it seems that metadata does not work with submodules

**Review by thewtex — APPROVED** [review]
@tbaudier thank you!

**Pull request merged**